TESTS/netsocket/README.md

```diff
@@ -1654,6 +1669,311 @@ through a dedicated socket
+2.  Define a DNS call in callback function that uses the event queue (call_in_callback_cb_t). Callback function shall not delay callbacks; instead it shall handle those immediately.
+3.  Start thread and event queue.
+4.  Set DNS callback function using the `nsapi_dns_call_in_set()` call.
+5.  Call `gethostbyname_async()` in a row 6 times with a different host names. Host names shall not be found from cache.
+6.  Verify that last `gethostbyname_async()` operation is rejected since there is room only for 5 simultaneous operations.
+7.  Verify that callback is called with correct parameters 5 times.
+
+**Expected result:**
+
+Sixth `gethostbyname_async()` is rejected. At least for one operation, callback is called with NSAPI_ERROR_TIMEOUT value.
+
+### ASYNCHRONOUS_DNS_SIMULTANEOUS_REPEAT
+
+**Description:**
+
+Verify that simultaneous asynchronous DNS queries work correctly when repeated in sequence. Call `NetworkInterface::gethostbyname_async()` in a row 5 times with a different host names. Wait for all requests to complete and verify result. Repeat the procedure 100 times.
+
+**Preconditions:**
+
+1.  Network interface is initialised.
+2.  Network connection is up.
+
+**Test steps:**
+
+1.  Call `gethostbyname_async()` in a row 5 times with a different host names.
+2.  Verify that callback is called with correct parameters 5 times for first operation.
+3.  Repeat 100 times steps 1 to 2.
+
+**Expected result:**
+
+Callback is called with NSAPI_ERROR_OK and IP address 5 times for every repeat.
+
+### SYNCHRONOUS_DNS
+
+**Description:**
+
+Verify basic functionality of synchronous DNS. Call `NetworkInterface::gethostbyname()` with a valid host name and verify result.
+
+**Preconditions:**
+
+1.  Network interface is initialised.
+2.  Network connection is up.
+
+**Test steps:**
+
+1.  Call `gethostbyname()` with a valid host name.
+2.  Verify that address was resolved and return value was valid.
+
+**Expected result:**
+
+Return value is NSAPI_ERROR_OK and IP address is obtained from the function call.
+
+### SYNCHRONOUS_DNS_MULTIPLE
+
+**Description:**
+
+Verify basic functionality of synchronous DNS. Call `NetworkInterface::gethostbyname()` with a list of 6 host names and verify result.
+
+**Preconditions:**
+
+1.  Network interface is initialised.
+2.  Network connection is up.
+
+**Test steps:**
+
+1.  Call `gethostbyname()` with a list of 6 host names
+2.  Verify that each of the addresses was resolved and return value was valid.
+
+**Expected result:**
+
+Return value is NSAPI_ERROR_OK and IP addresses are obtained from the function call.
+
+### SYNCHRONOUS_DNS_CACHE
+
+**Description:**
+
+Verify that the caching of DNS results works correctly. Call `NetworkInterface::gethostbyname()` 5 times with the same host name and verify result after each request. For first request, cache shall not contain the host name. Verify that first request completes slower than the requests made after it (where the response is found from cache).
```

Comment:
"...same host name, and verify the result after each request. For the first request,"
Also, I don't know what the second sentence means.
"Verify the first request completes more slowly"